app/controllers/merchant/items_controller.rb

```diff
@@ -1,5 +1,70 @@
 class Merchant::ItemsController < Merchant::BaseController
   def index
-    @items = Merchant.find(params["merchant_id"]).items
+    user = User.find(session[:user_id])
+    @items = Merchant.find(user.merchant.id).items
+    @merchant = Merchant.find(user.merchant.id)
```

Comment:
We shouldn't be setting 2 instance variables in a controller action. Proper convention is to set one and us that inside your view/use models methods to get what you need. I would keep @merchant, then use @merchant.items to get items.